e2e/tests/uniswap_trade_test.rs

```diff
@@ -129,16 +129,16 @@ async fn test_with_ganache() {
     db.clear().await.unwrap();
     let event_updater = EventUpdater::new(gp_settlement.clone(), db.clone(), None);
 
-    let price_estimator = UniswapPriceEstimator::new(
+    let price_estimator = Arc::new(UniswapPriceEstimator::new(
         Box::new(PoolFetcher {
             factory: uniswap_factory.clone(),
             web3: web3.clone(),
             chain_id,
         }),
         HashSet::new(),
-    );
+    ));
     let fee_calculator = Arc::new(MinFeeCalculator::new(
-        Box::new(price_estimator),
+        price_estimator.clone(),
```

Comment:
Does this increases the ref count? I've also seen `Arc::clone(...)`, is it the same?.

Reply:
Yes it's the same thing. Either way we are calling `Clone::clone` (the `clone` function of the `Clone` trait). It is just using different syntax. `.clone()` is more commonly used.

Reply:
Actually this is the e2e test, so here it doesn't really matter if we use the same PriceEstimator or a different one (in reality the API and driver run in different processes). Can't hurt though to have these components sharable. I'm sure there will be more use cases soon.